### PR TITLE
doc: add documentation about regular expression search 

### PR DESCRIPTION
**Pull request by kenhys**
Closes: #125

### CODE REVIEW

File: doc/source/tutorial/storage.rst [hunk 1]
```diff
@@ -654,6 +654,46 @@ See `Groonga's select command document
...
+  -- | Mroonga |
+  -- | Rroonga |
+  -- +---------+
+  -- 3 rows in set (0.003 sec)
+
+By using :ref:`boolean-mode-pragma-ss` pragma, you can search the records which matches "\*roonga" with ``content @~ ".?ro+nga"``.
+``@~`` is a Groonga's operator which executes a regular expression search, so ``content @~ ".?ro+nga"`` matches ``Groonga``, ``Mroonga`` and ``Rroonga``.
+
+See `Groonga's regular expression document
+<http://groonga.org/ja/docs/reference/regular_expression.html#syntax>`_ for more syntax details.
```
Comment (kou) [inline doc/source/tutorial/storage.rst]: Could you English document?

File: doc/source/tutorial/storage.rst [hunk 2]
```diff
@@ -654,6 +654,46 @@ See `Groonga's select command document
...
+  CREATE TABLE diaries (
+    content text,
+    FULLTEXT INDEX content_index (content) COMMENT 'tokenizer "TokenRegexp", normalizer "NormalizerAuto"'
+  ) ENGINE=Mroonga DEFAULT CHARSET=utf8mb4;
+
+  INSERT INTO diaries VALUES ('Groonga');
+  INSERT INTO diaries VALUES ('Mroonga');
+  INSERT INTO diaries VALUES ('Rroonga');
+
+  SELECT * FROM diaries WHERE MATCH(content) AGAINST ('*SS content @~ ".?ro+nga"' IN BOOLEAN MODE);
```
Comment (kou) [inline doc/source/tutorial/storage.rst]: This pattern can't be evaluated by index.

File: doc/source/tutorial/storage.rst (hunk 1 above)
Comment (kenhys) [reply doc/source/tutorial/storage.rst]: Oh, it should be fixed.

File: doc/source/tutorial/storage.rst (hunk 2 above)
Comment (kenhys) [reply doc/source/tutorial/storage.rst]: I've rewrite example which use index

File: doc/locale/ja/LC_MESSAGES/tutorial.po [hunk 3]
```diff
@@ -651,6 +651,52 @@ msgstr ""
...
+"す。 ``@~`` はGroongaで正規表現検索するためのオペレーターです。``content @~ "
+"\"\\\\\\\\A/var/log/auth\"`` は ``/var/log/auth.log`` に前方一致 ( ``/var/"
+"log/auth`` )でマッチします。"
+
+msgid ""
+"See `Groonga's regular expression document <http://groonga.org/docs/"
+"reference/regular_expression.html#syntax>`_ for more syntax details."
+msgstr ""
+"構文の詳細は `Groongaの正規表現のドキュメント <http://groonga.org/docs/"
+"reference/regular_expression.html#syntax>`_ を確認してください。"
```
Comment (kou) [inline doc/locale/ja/LC_MESSAGES/tutorial.po]: It should be Japanese document.

Comment (kenhys) [reply doc/locale/ja/LC_MESSAGES/tutorial.po]: Thanks, I've fixed.

File: doc/source/tutorial/storage.rst [hunk 4]
```diff
@@ -654,6 +654,44 @@ See `Groonga's select command document
...
+  CREATE TABLE paths (
+    content text,
+    FULLTEXT INDEX content_index (content) COMMENT 'tokenizer "TokenRegexp", normalizer "NormalizerAuto"'
+  ) ENGINE=Mroonga DEFAULT CHARSET=utf8mb4;
+
+  INSERT INTO paths VALUES ('/usr/bin/groonga');
+  INSERT INTO paths VALUES ('/var/log/auth.log');
+  INSERT INTO paths VALUES ('/var/log/messages');
+
+  SELECT * FROM paths WHERE MATCH(content) AGAINST ('*SS content @~ "\\\\A/var/log/auth"' IN BOOLEAN MODE);
```
[the remark this answers is not in the document]
Comment (kenhys) [reply doc/source/tutorial/storage.rst]: Thanks, I've added above insert query to ensure that prefix search works

File: doc/source/tutorial/storage.rst [hunk 5]
```diff
@@ -654,6 +654,45 @@ See `Groonga's select command document
...
+  -- +-------------------+
+  -- | content           |
+  -- +-------------------+
+  -- | /var/log/auth.log |
+  -- +-------------------+
+  -- 1 row in set (0.024 sec)
+
+By using :ref:`boolean-mode-pragma-ss` pragma, you can search the records which matches ``/var/log/auth.log`` with ``content @~ "\\\\A/var/log/auth"``.
+``@~`` is a Groonga's operator which executes a regular expression search, so ``content @~ "\\\\A/var/log/auth"`` matches ``/var/log/auth.log`` because it begin with ``/var/log/auth`` (prefix search).
+
```
Comment (kou) [inline doc/source/tutorial/storage.rst]: How about describing why `/tmp/local/var/log/auth.log` isn't matched?

Comment (kenhys) [reply doc/source/tutorial/storage.rst]: It is better to describe the reason why it is not matched.

Comment (kenhys) [reply doc/source/tutorial/storage.rst]: I've updated description.

Comment (kenhys) [reply doc/source/tutorial/storage.rst]: I've pushed 2 more commits, but not reflected to this PR.

**Pull request merged**